compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4052,22 +4063,67 @@ public BLangNode transform(MatchStatementNode matchStatementNode) {
+    public BLangXMLSequenceLiteral createXmlSequence(TemplateExpressionNode expressionNode) {
+        BLangXMLSequenceLiteral xmlSequenceLiteral = (BLangXMLSequenceLiteral)
+                TreeBuilder.createXMLSequenceLiteralNode();
+        xmlSequenceLiteral.pos = getPosition(expressionNode);
+
+        Node lastNode = null;
+        List<Node> adjacentTextNodes = new ArrayList<>();
+        int xmlContentSize = expressionNode.content().size();
+
+        for (int index = 0; index < xmlContentSize; index++) {
+            Node childItem = expressionNode.content().get(index);
+            if (childItem.kind() == SyntaxKind.XML_TEXT || childItem.kind() == SyntaxKind.INTERPOLATION) {
+                adjacentTextNodes.add(childItem);
+                lastNode = childItem;
+                if (index != xmlContentSize - 1) {
+                    continue;
+                }
+            }
+            // Handle previous node if it was of xml:Text or interpolation type
+            if (lastNode != null && (lastNode.kind() == SyntaxKind.XML_TEXT ||
+                    lastNode.kind() == SyntaxKind.INTERPOLATION)) {
+                if (adjacentTextNodes.size() > 1) {
+                    // Adjacent XML Text Literals (contains interpolated items and xml:Text items) should be
+                    // concatenated together
+                    xmlSequenceLiteral.xmlItems.add((BLangExpression) createXMLTextLiteral(adjacentTextNodes));
+                } else {
+                    xmlSequenceLiteral.xmlItems.add(createXmlSingletonItem(lastNode));
+                }
+                adjacentTextNodes.clear();
+                // Identify if sequence ends with xml:Text type or interpolation type
+                if (lastNode.kind() == childItem.kind()) {
+                    continue;
+                }
+            }
+            // Handle current node that is not of xml:Text type or interpolation type
+            xmlSequenceLiteral.xmlItems.add(createXmlSingletonItem(childItem));
+            lastNode = childItem;
         }
-        switch (contentKind) {
+        return xmlSequenceLiteral;
+    }
+
+    public BLangExpression createXmlSingletonItem(Node xmlTypeNode) {
+        switch (xmlTypeNode.kind()) {
             case XML_COMMENT:
             case XML_PI:
             case XML_ELEMENT:
             case XML_EMPTY_ELEMENT:
-                return createExpression(expressionNode.content().get(0));
+                return createExpression(xmlTypeNode);
             default:
-                return createXMLLiteral(expressionNode);
+                return (BLangExpression) createXMLTextLiteral(xmlTypeNode);
+        }
+    }
+
+    public BLangNode createXmlTemplateLiteral(TemplateExpressionNode expressionNode) {
+        if (expressionNode.content().isEmpty()) {
+            return createXMLEmptyLiteral(expressionNode);
+        }
+        if (expressionNode.content().size() > 1) {
```

Comment:
I think it's more logical to order this, emtpy -> size == 1 -> size more than one